docs/wiki/[Examples]-Deploy-Output-Values.md

```diff
@@ -0,0 +1,116 @@
+## Overview
+
+This example code helps you to extract required Terraform output values, which allow you to export structured data about your resources. You can use this data to configure other parts of your infrastructure with automation tools, or as a data source for another Terraform workload deployment folder, Outputs necessary to share data from a root module to child module and vice versa.
+
+Here is example how you can configure output for Azure landing zones architecture using Terraform. 
+
+This is a good starting point to learn how to configuring the landing zone state output, learn how to effectively utilize them in workload terraform modules. 
+
+> IMPORTANT: Ensure the module version is set to the latest, and don't forget to run `terraform init` if upgrading to a later version of the module.
+
+In your Landing Zone example folder create outputs.tf file and copy below terraform code for landing zone state output
+
+### Landing Zone root module folder `outputs.tf` 
+``` hcl
+# The following script provides summary of all resources output.
+# Created by the enterprise_scale module, formatted to allow easy identification of the resource IDs as stored in the Terraform state.
+
+output "resource_ids" {
+  value = {
+    for module_name, module_output in {
+      enterprise_scale = module.enterprise_scale
+    } 
+    module_name => {
+      for resource_type, resource_instances in module_output :
+      resource_type => {
+        for resource_name, resource_configs in resource_instances :
+        resource_name => keys(resource_configs)
+      }
+    }
+  }
+}
```

Comment:
Although we use this output in the test environments to give us a simple way to extract all of the resource IDs in a useful summary, I'm not sure this it the right approach for an example like this as it limits the data being output, and puts it in a very specific format.

Behind each resource type `key`, the module actually outputs the native Terraform resource outputs in map format, where they `key` represents the resource type

Instead I would rather see an example showing how to extract all config data for a specific resource type.

For example:

```hcl
output "azurerm_virtual_network" {
  value = values(module.test_connectivity.azurerm_virtual_network.connectivity)
}
```

Will extract a list of configuration settings for each virtual network created by the module, which will look like:

```shell
azurerm_virtual_network = [
  {
    "address_space" = tolist([
      "10.101.0.0/22",
    ])
    "bgp_community" = ""
    "ddos_protection_plan" = tolist([])
    "dns_servers" = tolist([])
    "edge_zone" = ""
    "flow_timeout_in_minutes" = 0
    "guid" = "ce03d850-478e-4603-9c03-a4ea6b9f5a18"
    "id" = "/subscriptions/fa2fa118-a60d-4700-9ef1-fa02beeaaea5/resourceGroups/12345-connectivity-northeurope/providers/Microsoft.Network/virtualNetworks/12345-hub-northeurope"
    "location" = "northeurope"
    "name" = "12345-hub-northeurope"
    "resource_group_name" = "12345-connectivity-northeurope"
    "subnet" = toset([])
    "tags" = tomap({
      "deployedBy" = "terraform/azure/caf-enterprise-scale/test_framework"
    })
    "timeouts" = null /* object */
  },
  {
    "address_space" = tolist([
      "10.100.0.0/22",
    ])
    "bgp_community" = ""
    "ddos_protection_plan" = tolist([])
    "dns_servers" = tolist([])
    "edge_zone" = ""
    "flow_timeout_in_minutes" = 0
    "guid" = "e535b12d-f92d-4e51-ab1f-09cc63681d8b"
    "id" = "/subscriptions/fa2fa118-a60d-4700-9ef1-fa02beeaaea5/resourceGroups/12345-connectivity-westeurope/providers/Microsoft.Network/virtualNetworks/12345-hub-westeurope"
    "location" = "westeurope"
    "name" = "12345-hub-westeurope"
    "resource_group_name" = "12345-connectivity-westeurope"
    "subnet" = toset([
      {
        "address_prefix" = "10.100.0.0/24"
        "id" = "/subscriptions/fa2fa118-a60d-4700-9ef1-fa02beeaaea5/resourceGroups/12345-connectivity-westeurope/providers/Microsoft.Network/virtualNetworks/12345-hub-westeurope/subnets/AzureFirewallSubnet"
        "name" = "AzureFirewallSubnet"
        "security_group" = ""
      },
    ])
    "tags" = tomap({
      "deployedBy" = "terraform/azure/caf-enterprise-scale/test_framework"
    })
    "timeouts" = null /* object */
  },
]
```

If you want to make this more useful for a customer to easily grab a network by type, you would want to make sure this is handled as a map, using a unique attribute to generate the key.

As we know the current key (the resource ID) isn't too friendly, but that we only allow a single virtual network hub per location, you might offer an example like the following:

```hcl
output "azurerm_virtual_network" {
  value = {
    for k,v in module.test_connectivity.azurerm_virtual_network.connectivity:
      v.location => v
    }
}
```

This would allow a customer to then reference a virtual network by location and extract all of the regular resource outputs and greatly simplify the examples you give below.

For example, the `local` for `connectivity_hub_vnet_name` would become:

```hcl
connectivity_hub_vnet_name = data.terraform_remote_state.eslz.outputs.azurerm_virtual_network.enterprise_scale.azurerm_virtual_network.westeurope.name
```